src/snowcli/cli/project/definition_manager.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import List
+import functools
+from typing import Optional
+from snowcli.exception import MissingConfiguration
+from snowcli.cli.project.definition import load_project_definition
+
+
+class DefinitionManager:
+    BASE_CONFIG_FILENAME = "snowflake.yml"
+    USER_CONFIG_FILENAME = "snowflake.local.yml"
+
+    project_root: Path
+    _project_config_paths: List[Path]
+
+    def __init__(self, project: str = "") -> None:
```

Comment:
It's a good practice to avoid mutable defaults. Consider using `project: str | None = None`